iceoryx_utils/include/iceoryx_utils/cxx/forward_list.hpp

```diff
@@ -0,0 +1,375 @@
+#include <cstdint>
+
+namespace iox
+{
+namespace cxx
+{
+/// @brief  C++11 compatible uni-directional forward list implementation.
+///         Adjustments in the API were done to not use exceptions and serve the requirement of
+///         a data structure movable over shared memory.
+///         attempt to add elements to a full list will be ignored.
+///         CAPACITY must at least be 1, (unintended) negative initialization is rejected with compile assertion
+///         limitation: concurrency concerns have to be handled by client side.
+/// @param T type user data to be managed within list
+/// @param CAPACITY number of maximum list elements a client can push to the list. minimum value is '1'
+template <typename T, int64_t CAPACITY>
+class forward_list
+{
+  private:
+    // forward declarations, private
+    struct ListNodeType;
+
+  public:
+    // forward declarations, public
+    class const_iterator;
+
+    static_assert(CAPACITY > 0, "CAPACITY must be an unsigned integral type >0");
+
+    using value_type = T;
+    using sizeType = typename std::make_unsigned<decltype(CAPACITY)>::type;
+
+    /// @brief constructor for an empty list (of T-types elements)
+    forward_list() noexcept;
+
+    /// @brief destructs the list and also calls the destructor of all
+    ///         contained elements
+    ~forward_list();
+
+    /// @brief copy constructor list including elements
+    /// @param[in] rhs is the list to copy from (same capacity)
+    forward_list(const forward_list& rhs) noexcept;
+
+    /// @brief move constructor list including elements
+    /// @param[in] rhs is the list to move-construct elements from (same capacity)
+    forward_list(forward_list&& rhs) noexcept;
+
+    /// @brief copy assignment, each element is copied (added) to the constructed list
+    ///         any existing elements in 'this'/lhs are removed (same behaviour as std::list : Assigns new contents to
+    ///         the container, replacing its current contents, and modifying its size accordingly.)
+    /// @param[in] rhs is the list to copy from (same capacity)
+    /// @return reference to created list
+    forward_list& operator=(const forward_list& rhs) noexcept;
+
+    /// @brief move assignment, list is cleared and initialized, elements are moved from source list
+    ///         any existing elements in 'this'/lhs are removed (same behaviour as std::list : Assigns new contents to
+    ///         the container, replacing its current contents, and modifying its size accordingly.)
+    /// @param[in] rhs is the list to move from ('source', same capacity)
+    /// @return reference to created list
+    forward_list& operator=(forward_list&& rhs) noexcept;
+
+    // cpy c'tor by different CapacityOther-sized list, repeat for remaining copy/move ops
+    // template<uint64_t CapacityOther, typename=std::enable_if<CapacityOther <Capacity, void>::type>  // PRQA S 1051
+    // list(const list<T, CapacityOther> & rhs) noexcept;  // PRQA S 1051
```

Comment:
Please do not commit out-commented code. Also please add a PRQA comment ;-)

Reply:
I do not see a reason for some kind of deviation here, what kind of rule is violated? I also do not understand what the enable_if is supposed to do, what exactly should the enabling condition be?